swift/ql/lib/codeql/swift/frameworks/Heuristic.qll

```diff
@@ -0,0 +1,48 @@
+/**
+ * Provides heuristic models that match taint sources and flow through unknown
+ * classes and libraries.
+ */
+
+import swift
+private import codeql.swift.dataflow.DataFlow
+private import codeql.swift.dataflow.FlowSources
+
+/**
+ * An imprecise flow source for an initializer call with a "contentsOf"
+ * argument that appears to be remote. For example:
+ * ```
+ * let myObject = MyClass(contentsOf: url)
+ * ```
+ */
+private class InitializerContentsOfRemoteSource extends RemoteFlowSource {
+  InitializerContentsOfRemoteSource() {
+    exists(InitializerCallExpr ce, Argument arg |
+      ce.getAnArgument() = arg and
+      arg.getLabel() = ["contentsOf", "contentsOfFile", "contentsOfPath", "contentsOfDirectory"] and
+      arg.getExpr().getType().getUnderlyingType().getName() = ["URL", "NSURL"] and
+      this.asExpr() = ce
```

Comment:
Any reason not to have a common implementation for this snippet that we can use in `InitializerContentsOfLocalSource`?

Reply:
Good idea, that will improve maintainability...

Reply:
Updated.